ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1658,6 +1752,107 @@ private void onInvalidRange(Exception exception) {
       numChunksRetrieved.set(0);
     }
   }
+
+  private class CachedFirstChunk extends FirstGetChunk {
+
+    /**
+     * Construct a CachedFirstChunk and initialize it with the {@link BlobMetadata} of the overall operation.
+     * @param blobMetadata Cached blob metadata for composite blobs
+     */
+    CachedFirstChunk(BlobMetadata blobMetadata) {
+      super();
+      /**
+       * If cache hit, extract all relevant metadata and save in appropriate variables.
+       * If the entry gets evicted later, we are still safe as we have saved all the metadata.
+       */
+      blobType = BlobType.MetadataBlob;
+      compositeBlobInfo = blobMetadata.getCompositeBlobInfo();
+      totalSize = compositeBlobInfo.getTotalSize();
+      chunkMetadataList = compositeBlobInfo.getChunkMetadataList();
+      blobInfo = blobMetadata.getBlobInfo();
+      lifeVersion = blobInfo.getLifeVersion();
+      serverBlobProperties = blobInfo.getBlobProperties();
+      chunkIndexToBuf = new ConcurrentHashMap<>();
+      chunkIndexToBufWaitingForRelease = new ConcurrentHashMap<>();
+      successfullyDeserialized = true;
+      logger.info(String.format("Metadata cache-hit for blobId = %s", blobId));
+    }
+
+    /**
+     * Return the {@link MessageFormatFlags} to associate with the first getBlob chunk operation.
+     * @return {@link MessageFormatFlags#Blob} for {@link GetBlobOptions.OperationType#Data}, or
+     *         {@link MessageFormatFlags#All} by default.
+     */
```

Comment:
we should update the comments here. Let's point out that for cached metadata chunk, we should request blob info from server to verify the existence of this chunk and update TTL.

Reply:
good catch!